pkg/tools/etcd_helper_watch.go

```diff
@@ -250,6 +259,10 @@ func (w *etcdWatcher) translate() {
 			return
 		case res, ok := <-w.etcdIncoming:
 			if ok {
+				if curLen := len(w.etcdIncoming); curLen > 0 {
```

Comment:
I'd suggest basic edge triggering then.  Log once when it crosses from zero to non-zero, and once when it crosses back again.